wordpress-mobile/WordPress-iOS
# Transitioned Period Chart to use live data

### Description
Fixes #11069.

To test:

- Checkout the branch & confirm that existing tests pass.
- Navigate to Stats for a given site and select a period in the filter bar (e.g., Days, Weeks, Months, Years).
- Observe that the chart now reflects live data, and updates when you select (by tapping) a dimension like Views, Visitors, Likes, or Comments.
- Rinse & repeat for a couple of different device types & orientations.
- Please keep in mind the caveats below. I've included some screenshots as well, but there should be no real aesthetic difference from #11210.

Update release notes:


### Notes

- The charts in the screenshots below include 10 bars; I think this can be attributed to [this default value](https://github.com/wordpress-mobile/WordPressKit-iOS/blob/develop/WordPressKit/StatsServiceRemoteV2.swift#L198). I intend to fix this as part of #11671. We might find that a change to `WordPressKit` is in order, but I'm hoping we can just specify a non-default value [here](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Stores/StatsPeriodStore.swift#L307).
- The chart does not currently update or really interact in any way with the Date bar. This will also be accounted for via #11671.
- You might wonder why we pass arrays of `BarChartDataConvertible` to the `PeriodChart` in `SiteStatsPeriodViewModel`. This accommodates the filter bar instance for Views, Visitors, Likes & Comments that resides in our `OverviewCell`. That is to say, a `FilterTabBar` selection essentially changes our data source index.
- On the bright side, no changes were made in this PR (or #11605 or #11616)  to `StatsBarChartView`. That suggests that we probably got the abstractions right.
- Peril has expressed a concern about the LOC changed, but a fair bit of the changes in this PR are deletions.

_iPhone XS Max (Portrait):_
<img width="584" alt="11069a" src="https://user-images.githubusercontent.com/221062/57495269-40ccf680-7282-11e9-930f-866f9bed2af6.png">

_iPhone XS Max (Landscape):_
<img width="1048" alt="11069b" src="https://user-images.githubusercontent.com/221062/57495270-40ccf680-7282-11e9-828b-8368c10f816f.png">
